File: py_coding_q_and_a/10.py
# flatten array

arr=[1,[2,3,4],5,[6,7,8,9]]

def flattenArr(arr):
    res=[]
    for x in arr:
        if type(x) != list:
            res.append(x)
        else:
            res+=x
    return res


print(flattenArr(arr))

File: js_coding_q_and_a/10.js
// Flatten a nested array

var arr = [1, 2, 3, 2, 3, 4, [1, 2, 3, 4, 5], [5, 6, 7, 8, 7, 8]];

const flatt = (arr) => {
  var res = [];
  arr.forEach((item) => {
    if (typeof item !== "object") {
      res.push(item);
    } else {
      res.push(...item);
    }
  });
  return res;
};

console.log(flatt(arr));
